authoring-recipes/types-of-recipes.md

```diff
@@ -0,0 +1,88 @@
+# Types of recipes
+
+Before you jump into coding recipes, it's important to figure out what type of recipe you're going to make.
+
+OpenRewrite offers support for three different types of recipes: declarative, refaster, and imperative. Each of these has its own advantages and disadvantages depending on the type of recipe you want to build. 
+
+In this doc, we'll briefly describe each of these so that you can determine what type of recipe works best for your needs. 
+
+## Declarative recipes
+
+[Declarative recipes](/reference/yaml-format-reference.md) are the simplest and most common recipes. They are entirely written in YAML, and they generally tie together existing recipes while adding some light configuration to them.
+
+The [common static analysis recipe](https://github.com/openrewrite/rewrite-static-analysis/blob/main/src/main/resources/META-INF/rewrite/common-static-analysis.yml) is a great example of this. It takes a bunch of simple recipes relating to static analysis and combines them into one larger recipe that can be run more easily. It does not add any logic, and it doesn't change the existing recipes.
+
+```yaml
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.staticanalysis.CommonStaticAnalysis
+displayName: Common static analysis issues
+description: Resolve common static analysis issues discovered through 3rd party tools.
+recipeList:
+  - org.openrewrite.staticanalysis.AtomicPrimitiveEqualsUsesGet
+  - org.openrewrite.staticanalysis.BigDecimalRoundingConstantsToEnums
+  - org.openrewrite.staticanalysis.BooleanChecksNotInverted
+  - org.openrewrite.staticanalysis.CaseInsensitiveComparisonsDoNotChangeCase
+  - org.openrewrite.staticanalysis.CatchClauseOnlyRethrows
+  - org.openrewrite.staticanalysis.ChainStringBuilderAppendCalls
+  # ...
+---
+```
+
+{% hint style="success" %}
+As a best practice, if your recipe can be declarative (meaning it can be built out of other recipes), then you should make it declarative.
+{% endhint %}
+
```

Comment:
Perhaps it's good to add a second example here where we actually add some configuration to achieve common migration steps, like renaming a class, a method, package or dependency. These are often needed for both OSS libraries and frameworks, as well as company internal frameworks. An example can be found here:
https://github.com/openrewrite/rewrite-spring/blob/main/src/main/resources/META-INF/rewrite/apache-httpclient-5.yml